packages/mds-compliance-engine/batch_process.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2021 City of Los Angeles
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import cache from '@mds-core/mds-agency-cache'
+import { ComplianceServiceClient, ComplianceSnapshotDomainModel } from '@mds-core/mds-compliance-service'
+import db from '@mds-core/mds-db'
+import logger from '@mds-core/mds-logger'
+import { ProcessManager, SerializedBuffers } from '@mds-core/mds-service-helpers'
+import { ModalityPolicy, ModalityPolicyTypeInfo } from '@mds-core/mds-types'
+import { minutes, now } from '@mds-core/mds-utils'
+import { processPolicy } from './engine'
+import { getAllInputs, getSupersedingPolicies } from './engine/helpers'
+
+const BATCH_SIZE = Number(process.env.BATCH_SIZE) || 5
+
+async function batchComplianceSnapshots(snapshots: ComplianceSnapshotDomainModel[]) {
+  let count = 0
+  const computeSnapshotPromises: Promise<SerializedBuffers<ComplianceSnapshotDomainModel>[]>[] = []
+  while (count < snapshots.length) {
+    const batch = snapshots.slice(count, count + BATCH_SIZE)
+    computeSnapshotPromises.push(ComplianceServiceClient.createComplianceSnapshots(batch))
+    count += BATCH_SIZE
+  }
+  await Promise.all(computeSnapshotPromises)
+}
+
+async function computeSnapshot() {
+  await cache.startup()
```

Comment:
nit: Mind adding a comment that we don't need to start the db here because `mds-db` has a lazy init? I know that, but when I was reading this I forgot 😭